tests/virt_autotest/kubevirt_tests_server.pm

```diff
@@ -0,0 +1,255 @@
+
+    # Create registries ready
+    script_run("cat > /etc/rancher/rke2/registries.yaml <<__END
+mirrors:
+  registry.suse.de:
+    endpoint:
+      - http://registry.suse.de
+configs:
+  registry.suse.de:
+    tls:
+      insecure_skip_verify: true
+__END");
+
+    assert_script_run("scp /etc/rancher/rke2/registries.yaml root\@$agent_ip:/etc/rancher/rke2/registries.yaml");	
+    mutex_create('CREATE_REGISTRIES_READY');
+
+    # Restart RKE2 service and check the service is active well after restart
+    systemctl('restart rke2-server.service');
+    check_service_status();
+}
+
+sub check_service_status {
+    # Check RKE2 services & kubevirt test environment
+    systemctl('is-active rke2-server.service');
+}
+
+sub deploy_kubevirt_manifests {
+    my $os_version = script_output("cat /etc/os-release | grep VERSION= | awk -F\" '{print \$2}'");
+    zypper_call("ar http://download.suse.de/ibs/Devel:/Virt:/SLE-" . $os_version . "/SUSE_SLE-" . $os_version . "_Update_standard/ devel");
+    zypper_call('ref devel');
+    zypper_call('in -f -r devel kubevirt-manifests kubevirt-virtctl kubevirt-tests containerized-data-importer-manifests') if (script_run('rpmquery kubevirt-manifests kubevirt-virtctl kubevirt-tests containerized-data-importer-manifests'));
+    assert_script_run("rpm -qa |grep -E \"virt-test|kubevirt|containerized\"");
+
+    assert_script_run("kubectl apply -f /usr/share/cdi/manifests/release/cdi-operator.yaml");
+    assert_script_run("kubectl apply -f /usr/share/cdi/manifests/release/cdi-cr.yaml");
+    assert_script_run("kubectl apply -f /usr/share/kube-virt/manifests/release/kubevirt-operator.yaml");
+    assert_script_run("kubectl apply -f /usr/share/kube-virt/manifests/release/kubevirt-cr.yaml");
+
+    assert_script_run("losetup -a -l"); # check all loop device to see if they refer to deleted files
+    assert_script_run("losetup -D"); # detach all loop device , the disk-image-provider needs to use it to setup images
+    assert_script_run("rm /tmp/hostImages /mnt/local-storage -r"); # remove existing local disks
+    assert_script_run("kubectl apply -f /usr/share/kube-virt/manifests/testing/rbac-for-testing.yaml");
+    assert_script_run("kubectl apply -f /usr/share/kube-virt/manifests/testing/disks-images-provider.yaml");
+    assert_script_run("kubectl apply -f /usr/share/kube-virt/manifests/testing/local-block-storage.yaml");
+
+    assert_script_run("ls /tmp/hostImages/ -l -R");
+    assert_script_run("ls /mnt/local-storage -R -l");
+    assert_script_run("losetup /dev/loop0");
+    assert_script_run("losetup -l -a");
+}
+
+sub run_virt_tests {
+    # Run virt-tests command for each go test files
+    my $cfg = new Config::Simple(data_url("virt_autotest/virt-tests.conf"));
+    my $section;
+    my $key;
+    my ($go_test, $skip_test, $extra_opt);
+    my ($artifacts, $junit_xml, $test_log);
+
+    foreach my $blocks (@kubevirt_tests) {
+        $section = $cfg->get_block($blocks);
```

Comment:
In cfg, is there any method like $cfg->get_all_block, so that you can skip configuring @kubevirt_tests?

Reply:
Unfortunately, no such method for use. https://metacpan.org/pod/Config::Simple